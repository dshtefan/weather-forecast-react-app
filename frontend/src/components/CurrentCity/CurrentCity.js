import React  from 'react';
import './CurrentCity.scss';
import WeatherIcon from "../WeatherIcon";
import WeatherInfo from "../WeatherInfo";
import Spinner from "../Spinner";

const CurrentCity = ({ city, loading }) => (
  loading ?
    <Spinner /> :
    city.error ?
      <div>{city.error}</div> :
      <div className="current-city">
        <div className="current-city__name">
          <div className="current-city__city-name">{city.city}</div>
          <div className="current-city__weather">{city.main}</div>
        </div>
        <div className="current-city__temp">
          <div className="current-city__weather-icon">
            <WeatherIcon iconNumber={city.icon} />
          </div>
          <div className="current-city__temperature">{city.temp}Â°</div>
        </div>
        <div className="current-city__info">
          <WeatherInfo {...city} />
        </div>
      </div>
);

export default CurrentCity;