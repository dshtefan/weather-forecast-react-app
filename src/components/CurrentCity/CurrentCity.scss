.current-city {
  grid-area: cd;
  max-height: 650px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 30px;
}
.current-city__name {

}
.current-city__city-name {
  font-size: 43px;
}
.current-city__weather {
  font-size: 26px;
}
.current-city__temp {
  display: flex;
}
.city-default__weather-icon img{
  width: 100px;
}
.current-city__temperature {
  font-size: 80px;
  color: #C0E7E2;
  margin-left: 20px;
}