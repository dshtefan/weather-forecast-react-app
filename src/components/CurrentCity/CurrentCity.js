import React  from 'react';
import './CurrentCity.scss';
import { Spinner, WeatherInfo } from "../";

const CurrentCity = ({ city, loading }) => (
  loading ?
    <Spinner /> :
    city.error ?
      <div>{city.error}</div> :
      <div className="current-city">
        <div className="current-city__name">
          <div className="current-city__city-name">{city.city}</div>
          <div className="current-city__weather">{city.main}</div>
        </div>
        <div className="current-city__temp">
          <div className="current-city__weather-icon">
            <img src={`https://dshtefan.github.io/weather-search-2/icons/${city.icon}.svg`} alt="" />
          </div>
          <div className="current-city__temperature">{city.temp}Â°</div>
        </div>
        <div className="current-city__info">
          <WeatherInfo {...city} />
        </div>
      </div>
);

export default CurrentCity;