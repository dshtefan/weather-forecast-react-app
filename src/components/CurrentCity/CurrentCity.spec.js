import React from 'react';
import CurrentCity from './CurrentCity';

describe('CurrentCity', () => {
  const cityError = { error: 'cityError' };
  const city = {
    city: 'Moscow',
    main: 'Rain',
    icon: '10',
    temp: '2',
    pressure: '997',
    humidity: '86',
    wind: '5',
    lat: '60',
    lon: '30'
  };
  const props = {
    city,
    loading: true
  };

  describe('загрузка', () => {
    const currentCity = shallow(<CurrentCity {...props}/>);
    it('отображается корректно', () => expect(shallowToJson(currentCity)).toMatchSnapshot());
  });

  describe('ошибка при получении города', () => {
    const nextProps = {
      ...props,
      loading: false,
      city: cityError
    };
    const currentCity = shallow(<CurrentCity {...nextProps} />);
    it('отображается корректно', () => expect(shallowToJson(currentCity)).toMatchSnapshot());
  });

  describe('отображение города', () => {
    const nextProps = {
      ...props,
      loading: false
    };
    const currentCity = shallow(<CurrentCity {...nextProps} />);
    it('отображается корректно', () => expect(shallowToJson(currentCity)).toMatchSnapshot());
  });

});