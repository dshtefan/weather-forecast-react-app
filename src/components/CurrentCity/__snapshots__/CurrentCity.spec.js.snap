// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCity загрузка отображается корректно 1`] = `<Spinner />`;

exports[`CurrentCity отображение города отображается корректно 1`] = `
<div
  className="current-city"
>
  <div
    className="current-city__name"
  >
    <div
      className="current-city__city-name"
    >
      Moscow
    </div>
    <div
      className="current-city__weather"
    >
      Rain
    </div>
  </div>
  <div
    className="current-city__temp"
  >
    <div
      className="current-city__weather-icon"
    >
      <WeatherIcon
        iconNumber="10"
      />
    </div>
    <div
      className="current-city__temperature"
    >
      2
      °
    </div>
  </div>
  <div
    className="current-city__info"
  >
    <WeatherInfo
      city="Moscow"
      humidity="86"
      icon="10"
      lat="60"
      lon="30"
      main="Rain"
      pressure="997"
      temp="2"
      wind="5"
    />
  </div>
</div>
`;

exports[`CurrentCity ошибка при получении города отображается корректно 1`] = `
<div>
  cityError
</div>
`;
